In [2]:
a,b = "56"
s = f'{a*3:9}|{b*5:^9}|{a*7:>9}'
print(s)

555      |  66666  |  5555555


Formatting:

f'{a*3:9}': by default left-aligns strings in width 9
→ '555 '

f'{b*5:^9}': center-align in width 9
→ ' 66666 '

f'{a*7:>9}': right-align in width 9
→ ' 5555555'

In [7]:
type(print(s.upper().split("-")))

['555      |  66666  |  5555555']


NoneType

| Expression                          | Type                                 |
| ----------------------------------- | ------------------------------------ |
| `print(s.upper().split("-"))`       | `NoneType` (if evaluated as a value) |
| `type(s.upper().split("-"))`        | `list`                               |
| `type(print(s.upper().split("-")))` | `NoneType`                           |


In [8]:
type(s[:3].upper().split())

list

In [9]:
type(s.islower().isapha())

AttributeError: 'bool' object has no attribute 'isapha'

 Final conclusion:
The expression s.islower().isapha() (or .isalpha()) is invalid and will raise an error.

So the type of the expression itself does not exist because it raises an exception.

If you meant s.islower(), its type is bool.

In [11]:
type(s.startswith())

TypeError: startswith expected at least 1 argument, got 0

What does startswith() do?
startswith() is a string method used to check if the string starts with a specified prefix.

It requires at least one argument: the prefix to check for.

s.startswith('a')

What happens if you call s.startswith() with no arguments?
Python will raise a TypeError because the required positional argument is missing.

So:
The expression s.startswith() without any arguments will raise an error.

Therefore, it does not have a type because it does not successfully evaluate.

| Expression                | Result                                   |
| ------------------------- | ---------------------------------------- |
| `s.startswith()`          | Raises `TypeError` (missing argument)    |
| `type(s.startswith('a'))` | `<class 'bool'>` (returns True or False) |


If you want to know the type when used correctly:

type(s.startswith('prefix'))  # returns <class 'bool'>


In [12]:
type(list(s))

list

In [13]:
print(s)

555      |  66666  |  5555555


What is the value of the expression bool('0.0')?

Key Concept:
'0.0' is a non-empty string.

In Python, any non-empty string passed to bool() evaluates to True.

It doesn't matter that the string represents the number zero — it is still a non-empty string.

✅ Final Answer:
True


What is the value of the expression bool(1 and 2 and 0)?

Step-by-step:
1. and in Python:
The and operator returns the first falsy value it encounters, or the last value if all are truthy.

Expression: 1 and 2 and 0
1 is truthy → continue

2 is truthy → continue

0 is falsy → so the expression evaluates to 0

2. bool(0):
0 is falsy → bool(0) is False

✅ Final Answer:
False
